R/arc-select.R

```diff
@@ -454,10 +450,91 @@ count_results <- function(req, query) {
   resp <- httr2::resp_body_string(
     httr2::req_perform(
       httr2::req_url_query(n_req, f = "json"),
-      error_call = rlang::caller_env()
+      error_call = error_call
     )
   )
 
   RcppSimdJson::fparse(resp)[["count"]]
 }
 
+#' Set page size and check page size validity
+#'
+#' @noRd
+set_page_size <- function(
+    page_size = NULL,
+    max_records = 25000L,
+    error_call = rlang::caller_env()
+    ) {
+  # if its null, just use max records (default)
+  if (is.null(page_size)) {
+    return(max_records)
+  }
+
+  # ensure its an integer.
+  page_size <- as.integer(page_size)
+  page_size_len <- length(page_size)
+
+  if (page_size_len > 1) {
+    cli::cli_abort(
+      "{.arg page_size} must be length 1 not {page_size_len}",
+      call = error_call
+    )
+  }
+
+  if (page_size < 1) {
+    cli::cli_abort(
+      "{.arg page_size} must be a positive integer.",
+      call = error_call
+    )
+  }
+
+  if (page_size > max_records) {
+    cli::cli_abort(
+      "{.arg page_size} ({page_size}) cannot excede layer's {.field maxRecordCount} property ({max_records})",
+      call = error_call
+    )
+  }
+
+  page_size
+}
+
+
+
+#' Bind list with parsed ESRI geometry into sf object
+#'
+#' @noRd
+rbind_results <- function(res, crs = NULL, error_call = rlang::caller_env()) {
+  empty_res <- vapply(res, rlang::is_empty, FALSE)
+
+  if (all(empty_res)) {
+    cli::cli_alert_info("No features returned from query")
+    return(data.frame())
+  }
+
```

Comment:
Do you have a repro for this scenario? We should have a test for it. Because I feel like I fixed a related issue with `parse_esri_json()` not too long ago! https://github.com/R-ArcGIS/arcgisutils/pull/21